src/assets/scripts/peer.js

```diff
@@ -0,0 +1,90 @@
+import SimplePeer from 'simple-peer';
+import Debug from 'debug';
+
+const peerStore = [];
+
+function signalPeer(peerId, signalData) {
+  const peer = peerStore.find(peer => peer.id === peerId);
+  if (!peer) {
+    throw new Error(`Peer "${peerId}" not found`);
+  }
+  peer.signal(signalData);
+}
+
+class Deferred {
```

Comment:
I'm not sure if we will end up using it, but the `Deferred` model is very useful, maybe it should be pulled out.